File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        # dictionary to store the nums
        nums_dict = dict()

        # iterating over the nums
        for i in range(len(nums)):

            # calculate the complement for the current num
            complement = target - nums[i]

            # check if present in dict or not
            if complement in nums_dict:
                # if present, then return it and current num
                return [i, nums_dict[complement]]

            else:
                # if not present, add it to dict
                nums_dict[nums[i]] = i
            




        # for i in range(len(nums)):
        #     for j in range(i+1, len(nums)):
        #         if nums[i] + nums[j] == target:
        #             return [i, j]
        

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {


        int arr[] = new int[2];
        for(int i=0; i<nums.length-1; i++){
            for(int j=i+1; j<nums.length; j++){

                if(nums[i]+nums[j]==target){
                    arr[0] = i;
                    arr[1] = j;

                }

            }
        }
        return arr;
        
    }
}